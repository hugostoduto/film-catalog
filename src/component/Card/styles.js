import styled, { css } from 'styled-components';

export const Container = styled.div`
${({ theme }) => css`
  .card-container{
    width: 200px;
    height: 300px;
    display: flex;
    flex-direction: column;
    justify-content: center;

  }
  .card-title{

  }
  .card-name{

  }
  .card-year{

  }
`}
`;
