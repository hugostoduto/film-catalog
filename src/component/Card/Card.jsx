import * as Styled from './styles';
import React from 'react';

export const Card = ({ name, year, image, onClick }) => {
  return (
    <Styled.Container>
      <div onClick={onClick} className="card-container">
        <img src={image} alt={name} className="card cover" />
        <div className="card-title">
          <h4 className="card-name">{name}</h4>
          <p className="card-year">{year}</p>
        </div>
      </div>
    </Styled.Container>
  );
};

