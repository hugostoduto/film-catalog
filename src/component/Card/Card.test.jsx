import { fireEvent, render, screen } from '@testing-library/react';
import { Card } from './Card';

describe('<Card/>', () => {
  it('Should render a img', () => {
    render(<Card image="cover" />);
    const image = screen.getByRole('img', { image: /cover/i });
    expect(image).toBeInTheDocument();
  });
  it('Should render a heading with name', () => {
    render(<Card name="filme" />);
    const name = screen.getByRole('heading', { name: /filme/i });
    expect(name).toBeInTheDocument();
  });
  it('Should render a heading with year', () => {
    render(<Card year="2023" />);
    const year = screen.getByRole('heading', { year: /2023/i });
    expect(year).toBeInTheDocument();
  });
  it('Should call function onClick', () => {
    const fn = jest.fn();
    render(<Card name="filme" onClick={fn} />);

    const card = screen.getByRole('heading', { name: /filme/i });
    fireEvent.click(card);
    expect(fn).toHaveBeenCalledTimes(1);
  });
  it('should render Card correctly', () => {
    const fn = jest.fn();
    render(<Card name="nome" image="image.img" year="2023" onClick={fn} />);
    expect(screen.getByRole('img', { name: /nome/i })).toHaveAttribute('src');
  });
  it('Should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(
      <Card name="nome" image="image.img" year="2023" onClick={fn} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
