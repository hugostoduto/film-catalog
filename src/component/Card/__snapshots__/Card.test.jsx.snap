// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> Should match snapshot 1`] = `
.c0 .card-container {
  width: 200px;
  height: 300px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
>
  <div
    class="card-container"
  >
    <img
      alt="nome"
      class="card cover"
      src="image.img"
    />
    <div
      class="card-title"
    >
      <h4
        class="card-name"
      >
        nome
      </h4>
      <p
        class="card-year"
      >
        2023
      </p>
    </div>
  </div>
</div>
`;
